### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'Template.loginButtons')

### ISSUE THREAD

**Issue by mattiasewers** [jedwards1211/meteor-webpack-react#101]
Getting this error running dev, using react 0.14.

**Comment by mattiasewers** [comment]
Also, when running in production it seems that es7 decorators are not working...

**Comment by jbbr** [comment]
Try switching back to commit bfc71981a7008adf007e7b5a94c17f28a0a12adb or less - Or try the new https://github.com/jedwards1211/meteor-webpack-react/tree/webpack-meteor-tools branch.

Or: merge the PR's of ` haydnhkim` #91 and #98 

Currently master is broken due to incomplete SSR additions.

**Comment by jedwards1211** [comment]
The webpack-meteor-tools branch should have this problem as well -- I need to figure out a different hack for server side source maps/original files in the debugger.

**Comment by jedwards1211** [comment]
Also the first and so far only release shouldn't have this problem

**Comment by jedwards1211** [comment]
Okay, this is now fixed in the latest version of the branch: https://github.com/jedwards1211/meteor-webpack-react/tree/webpack-meteor-tools
I figured out a way to support server-side source maps without having to `require` in the server bundle.

**Comment by mattiasewers** [comment]
@jedwards1211 Alright, progress! Are you going to merge it soon?

**Comment by jedwards1211** [comment]
We'll see...I've been terribly busy lately

**Comment by jedwards1211** [comment]
I'm still confused how `?stage=0` fixes this problem, several people have reported that.  Do the login buttons actually show up in the webpage or does it simply not error out?

**Comment by jbbr** [comment]
@jedwards1211 It's easy:

Babel config in `webpack.config.dev.js`: stage=0, transform plugins ...
Babel config in `webpack.config.prod.js`: None
Babel config in common `webpack.config.js`: Default (stage=2)

decorators are stage 1 - so when build prod decorators did not work (used for `reactMixin`)

**Comment by jedwards1211** [comment]
Okay, I finally revisited this.  I was confused since I never use Blaze -- `Template` isn't available on the server side, so I got rid of the PR that monkeypatched it for the sake of SSR (which was ruining the client side anyway due to `var` hoisting, which I didn't think about.

In short, wrap all references to `Template` in `Meteor.isClient` checks.

Also, I revamped how the webpack configs are created to use `webpack-merge` and babel options in the `query` parameter as necessary.

Cheers!
